### ISSUE TITLE
Remove/make optional Column selection in Scikit-learn tools

### ISSUE THREAD

**Issue by jgoecks** [bgruening/galaxytools#725]
I'd like to run a Sk-learn workflow on a collection of datasets (e.g. preprocessing, training, evaluation). However, this isn't possible right now unless each dataset has the same number of columns. This is because column selection is mandatory for all training/test datasets. In order of preference, I suggest (1) moving to an approach where columns are not selected at all or (2) column selection is optional. Given that Galaxy can cut columns, I lean towards (1). This is a substantial change—what do others think? @bgruening @qiagu

**Comment by qiagu** [comment]
It will be a big change, great enhancement though. If going with (1), probably, we need a separate tool to do column selection or splitting? But this still seems to be a variant of (2). If we completely ignore the column selection, how to make the tool(s) work for a dataset containing both features and labels in columns?

**Comment by qiagu** [comment]
Can we give the column selector an option 'All'? I know some sklearn functions accept this option in addition to int and float.

**Comment by jgoecks** [comment]
Nm, I think we can make (2) work and preserve existing functionality. I am closing in favor of a future PR.
